Sources/Views/MusicProgressSliderCell.m

```diff
@@ -0,0 +1,55 @@
+//
+//  MusicProgressSlider.m
+//  Hermes
+//
+//  Created by Xinhong LIU on 19/4/15.
+//
+//
+
+#import "MusicProgressSliderCell.h"
+
+@implementation MusicProgressSliderCell
+
+@synthesize knobImage = _knobImage;
+
+- (NSImage *)knobImage {
+  return _knobImage;
+}
+
+- (void)setKnobImage:(NSImage *)knobImage {
+  _knobImage = knobImage;
+  if (_knobImage == nil) {
+    // default knob image
+    _knobImage = [NSImage imageNamed:@"knob.png"];
+  }
+}
+
+- (id)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self setKnobImage:[NSImage imageNamed:@"knob.png"]];
+  }
+  return self;
+}
+
+- (NSRect)knobRectFlipped:(BOOL)flipped {
+  float KNOB_WIDTH = [[self knobImage] size].width;
+  float KNOB_HEIGHT = [[self knobImage] size].height;
```

Comment:
In general local variables in obj-c should follow the convention of property names -- for example here you would be better with `knobWidth` and `knobHeight`.

If you want to remove the brackets you can also do `self.knobImage.size.width`, and so on.